### PR TITLE
Relicense under MIT and Apache-2.0

### PR DESCRIPTION
**Pull request by tom-borcin**
Relicensing the espflash project under MIT and Apache-2.0 licenses as agreed on #224

### CODE REVIEW

**Review by MabezDev — APPROVED** [review]
Looks good to me!

**Review by MabezDev — CHANGES_REQUESTED** [review]
Actually, can you remove the LICENSE file in `cargo-espflash` too?

**Review by MabezDev — CHANGES_REQUESTED** [review]
Sorry! Last thing I just remembered, could you change the license inside the [Cargo.toml](https://github.com/esp-rs/espflash/blob/d64ff68a925225c6ac7aae332c2ac32b0ccb7e3d/espflash/Cargo.toml#L12) of both espflash & cargo-espflash.

It should be possible to replace with `"MIT OR Apache-2.0"`.

File: LICENSE-APACHE [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+To apply the Apache License to your work, attach the following
+boilerplate notice, with the fields enclosed by brackets "[]"
+replaced with your own identifying information. (Don't include
+the brackets!) The text should be enclosed in the appropriate
+comment syntax for the file format. We also recommend that a
+file or class name and description of purpose be included on the
+same "printed page" as the copyright notice for easier
+identification within third-party archives.
+
+Copyright [yyyy] [name of copyright owner]
```
Comment (jessebraham) [inline LICENSE-APACHE:63]: The copyright year and owner have not been added here

Comment (tom-borcin) [reply LICENSE-APACHE:63]: Honestly I didn't know what to do with those placeholders. [rust-clippy](https://github.com/rust-lang/rust-clippy/blob/master/LICENSE-APACHE) has it filled out, but [rust](https://github.com/rust-lang/rust/blob/master/LICENSES/Apache-2.0.txt) and [cargo](https://github.com/rust-lang/cargo/blob/master/LICENSE-APACHE) does not.

From [Apache-2.0 license](https://www.apache.org/licenses/LICENSE-2.0) I understood that you add that declaration only to specific files and not to the repository as a whole. But I may be wrong, I am not a native speaker after all.

Comment (jessebraham) [reply LICENSE-APACHE:63]: Ahh okay, no worries then. I am not very knowledgable about this stuff and just noticed those placeholders 😁

**Review by MabezDev — APPROVED** [review]
LGTM, thank you Tomas!

**Comment by maximeborges** [comment]
The badge in the README.md is generated from the crates.io page, which is still referring to GPL-2.
Also could we have a new release with this new license (and the erase OTA flag :innocent: )

**Comment by jessebraham** [comment]
@maximeborges I am planning on publishing a new release next week, probably Monday or Tuesday. Just wanted a few days to test it out and tweak things as needed because there have been a lot of changes since the last release.

**Pull request merged**